src/sentry/migrations/0700_backfill_group_priority_to_self_hosted.py

```diff
@@ -0,0 +1,188 @@
+            group_id__in=[gv.id for gv in group_values_list]
+        ).values("group_id", "user_id", "team_id")
+    }
+
+    group_owner_map = {}
+
+    for group_owner in (
+        GroupOwner.objects.annotate(
+            position=Window(Rank(), partition_by=[F("group_id"), F("type")], order_by="-date_added")
+        )
+        .filter(position=1, group_id__in=[g.id for g in group_values_list])
+        .values("group_id", "user_id", "team_id", "type")
+    ):
+        group_owner_map[(group_owner["group_id"], group_owner["type"])] = group_owner
+
+    snapshots = []
+    for group_value in group_values_list:
+        assignee = group_assignee_map.get(group_value.id)
+        suspect_owner = group_owner_map.get((group_value.id, GroupOwnerType.SUSPECT_COMMIT.value))
+        ownership_owner = group_owner_map.get((group_value.id, GroupOwnerType.OWNERSHIP_RULE.value))
+        codeowners_owner = group_owner_map.get((group_value.id, GroupOwnerType.CODEOWNERS.value))
+        snapshot: GroupAttributesSnapshot = {
+            "group_deleted": False,
+            "project_id": group_value.project_id,
+            "group_id": group_value.id,
+            "status": group_value.status,
+            "substatus": group_value.substatus,
+            "first_seen": group_value.first_seen.strftime("%Y-%m-%dT%H:%M:%S.%fZ"),
+            "num_comments": group_value.num_comments,
+            "timestamp": datetime.now().isoformat(),
+            "assignee_user_id": assignee["user_id"] if assignee else None,
+            "assignee_team_id": assignee["team_id"] if assignee else None,
+            "owner_suspect_commit_user_id": suspect_owner["user_id"] if suspect_owner else None,
+            "owner_ownership_rule_user_id": ownership_owner["user_id"] if ownership_owner else None,
+            "owner_ownership_rule_team_id": ownership_owner["team_id"] if ownership_owner else None,
+            "owner_codeowners_user_id": codeowners_owner["user_id"] if codeowners_owner else None,
+            "owner_codeowners_team_id": codeowners_owner["team_id"] if codeowners_owner else None,
+        }
+        snapshots.append(snapshot)
+
+    return snapshots
+
+
+def bulk_send_snapshot_values(
+    group_ids: list[int],
+    Group: type[Group],
+    GroupAssignee: type[GroupAssignee],
+    GroupOwner: type[GroupOwner],
+) -> None:
+    group_list = []
+    if group_ids:
+        group_list.extend(_bulk_retrieve_group_values(group_ids, Group))
+
+    snapshots = _bulk_retrieve_snapshot_values(group_list, GroupAssignee, GroupOwner)
+
+    for snapshot in snapshots:
+        produce_snapshot_to_kafka(snapshot)
+
+
+def backfill_group_attributes_to_snuba(
+    apps: StateApps, schema_editor: BaseDatabaseSchemaEditor
+) -> None:
+    Group = apps.get_model("sentry", "Group")
+    GroupAssignee = apps.get_model("sentry", "GroupAssignee")
+    GroupOwner = apps.get_model("sentry", "GroupOwner")
+
+    backfill_key = "backfill_group_attributes_to_snuba_progress"
```

Comment:
I assume the key has expired by now, but maybe we should just slightly change it just in case?